src/analysis/interpolation/qgsgridfilewriter.cpp

```diff
@@ -56,37 +57,45 @@ int QgsGridFileWriter::writeFile( QgsFeedback *feedback )
 
   double currentYValue = mInterpolationExtent.yMaximum() - mCellSizeY / 2.0; //calculate value in the center of the cell
   double currentXValue;
-  double interpolatedValue;
 
+  // Builds the list of coords
   for ( int i = 0; i < mNumRows; ++i )
   {
     currentXValue = mInterpolationExtent.xMinimum() + mCellSizeX / 2.0; //calculate value in the center of the cell
+    QVector<QgsPointXY> points;
+    points.reserve( mNumColumns );
     for ( int j = 0; j < mNumColumns; ++j )
     {
-      if ( mInterpolator->interpolatePoint( currentXValue, currentYValue, interpolatedValue, feedback ) == 0 )
+      if ( feedback )
       {
-        outStream << interpolatedValue << ' ';
-      }
-      else
-      {
-        outStream << "-9999 ";
+        if ( feedback->isCanceled() )
+        {
+          outputFile.remove();
+          return 3;
+        }
+        feedback->setProgress( 100.0 * i / static_cast< double >( mNumRows ) );
       }
+      points.push_back( { currentXValue, currentYValue } );
       currentXValue += mCellSizeX;
     }
-    outStream << endl;
-    currentYValue -= mCellSizeY;
 
-    if ( feedback )
+
+    std::function<double( const QgsPointXY & )> interpolate = [ = ]( const QgsPointXY & point )
     {
-      if ( feedback->isCanceled() )
-      {
-        outputFile.remove();
-        return 3;
-      }
-      feedback->setProgress( 100.0 * i / static_cast< double >( mNumRows ) );
+      return mInterpolator->interpolatedPoint( point, feedback );
+    };
+
+    QFuture< double > interpolated = QtConcurrent::mapped( points, interpolate );
```

Comment:
As far as I can see there's no way to abort this early anymore?

Reply:
Why not? Feedback is captured in.

Reply:
but if you cancel, it'll still iterate over all remaining pixels right?  (just not doing much actual work for them). Would be nicer if the future was canceled as soon as the feedback is canceled...

Reply:
only one row actually, because the implementation processes one row at a time. I don't think this is a big issue.